recipes/Kili_fine_tuning_tutorial.ipynb

```diff
@@ -0,0 +1,2081 @@
+{
```

Comment:
<pre class="ql-syntax" spellcheck="false">&nbsp; index = len(returned_value)
&nbsp; for predefined_class in all_predefined_classes:
&nbsp;&nbsp;&nbsp; if predefined_class in returned_value.upper() and returned_value.upper().index(predefined_class) &lt; index:
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; result = predefined_class
</pre><p> </p><p>so in this part I recommend you to use a more advanced parser &amp; sanitizer for cleaning the openai prediction (you have to adapt the code to your notebook):</p><p> </p><pre class="ql-syntax" spellcheck="false">def parse_and_sanitize_openai_response(response: str, categories) -&gt; str:
&nbsp;&nbsp;&nbsp; if response in categories:
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; return response

&nbsp;&nbsp;&nbsp; response = response.replace("OUTPUT:", "")
&nbsp;&nbsp;&nbsp; if response in categories:
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; return response

&nbsp;&nbsp;&nbsp; response = response.strip()&nbsp; # remove spaces at the start/end
&nbsp;&nbsp;&nbsp; if response in categories:
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; return response

&nbsp;&nbsp;&nbsp; response = response.strip(".")
&nbsp;&nbsp;&nbsp; if response in categories:
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; return response

&nbsp;&nbsp;&nbsp; min_distance = 1e9
&nbsp;&nbsp;&nbsp; closest_category = None
&nbsp;&nbsp;&nbsp; for class_name in categories:
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; dist = Levenshtein.distance(class_name, response)
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; if dist &lt; min_distance:
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; closest_category = class_name

&nbsp;&nbsp;&nbsp; return closest_category
</pre>
<br/>

 _Reply via <a href='https://app.reviewnb.com/kili-technology/kili-python-sdk/pull/1259/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='23'/>

Reply:
<p>Will update this in version 2.0 of this NB</p>